### PR TITLE
Java 9 smoke testing now use a regular TestCoverage and testJavaHome

### PR DESCRIPTION
**Pull request by eskatos**
Change TC Java 9 smoke testing job to be a regular functional testing job and make use of `testJavaHome`.

### CODE REVIEW

**Comment by eskatos** [comment]
This PR goes with https://github.com/gradle/gradle/pull/3756

File: .teamcity/Gradle_Check/model/CIBuildModel.kt [hunk 1]
```diff
@@ -39,7 +39,7 @@ data class CIBuildModel (
                     functionalTests = listOf(
                             TestCoverage(TestType.quickFeedbackCrossVersion, OS.linux, JvmVersion.java7),
                             TestCoverage(TestType.quickFeedbackCrossVersion, OS.windows, JvmVersion.java7),
-                            TestCoverage(TestType.java9Smoke, OS.linux, JvmVersion.java8),
+                            TestCoverage(TestType.quick, OS.linux, JvmVersion.java9),
```
Comment (wolfs) [inline .teamcity/Gradle_Check/model/CIBuildModel.kt]: I think the `TestType` should be `platform`, given that the integration tests have been running with `forking`.

Comment (eskatos) [reply .teamcity/Gradle_Check/model/CIBuildModel.kt]: Right. `forking` is needed here.

`java9Smoke` didn't run tests though, only integration tests.
Shall we introduce a new `TestType` for this? or could we go with `platform` and also run tests on Java 9?

Comment (wolfs) [reply .teamcity/Gradle_Check/model/CIBuildModel.kt]: I wouldn't introduce a new `TestType`. Let's see if the unit tests pass on Java 9 (I fear some will fail...). We should just exclude whichever are failing and run unit tests on Java 9, too.

**Pull request merged**